cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2757 @@
+
+  /// @brief Returns the object’s timestamp (microseconds from the Unix epoch).
+  int64_t timestamp() const { return mgp::local_date_time_timestamp(ptr_); }
+
+  bool operator==(const LocalDateTime &other) const;
+  LocalDateTime operator+(const Duration &dur) const;
+  LocalDateTime operator-(const Duration &dur) const;
+  Duration operator-(const LocalDateTime &other) const;
+
+  bool operator<(const LocalDateTime &other) const;
+
+ private:
+  mgp_local_date_time *ptr_;
+};
+
+/// @brief Wrapper class for @ref mgp_duration.
+class Duration {
+ private:
+  friend class Date;
+  friend class LocalTime;
+  friend class LocalDateTime;
+  friend class Value;
+  friend class Record;
+  friend class Result;
+  friend struct Parameter;
+
+ public:
+  /// @brief Creates a Duration from the copy of the given @ref mgp_duration.
+  explicit Duration(mgp_duration *ptr) : ptr_(mgp::duration_copy(ptr, memory)) {}
+  /// @brief Creates a Duration from the copy of the given @ref mgp_duration.
+  explicit Duration(const mgp_duration *const_ptr)
+      : ptr_(mgp::duration_copy(const_cast<mgp_duration *>(const_ptr), memory)) {}
+
+  /// @brief Creates a Duration object from the given string in the following format: `P[nD]T[nH][nM][nS]`, where (1)
+  /// `n` stands for a number, (2) capital letters are used as a separator, (3) each field in `[]` is optional, and (4)
+  /// only the last field may be a non-integer.
+  explicit Duration(std::string_view string) : ptr_(mgp::duration_from_string(string.data(), memory)) {}
+
+  /// @brief Creates a Duration object from the given number of microseconds.
+  explicit Duration(int64_t microseconds) : ptr_(mgp::duration_from_microseconds(microseconds, memory)) {}
+
+  /// @brief Creates a Duration object with the given `day`, `hour`, `minute`, `second`, `millisecond`, and
+  /// `microsecond` properties.
+  Duration(double day, double hour, double minute, double second, double millisecond, double microsecond);
+
+  Duration(const Duration &other) : Duration(other.ptr_) {}
+  Duration(Duration &&other) : ptr_(other.ptr_) { other.ptr_ = nullptr; };
+
+  Duration &operator=(const Duration &other) = delete;
+  Duration &operator=(Duration &&other) = delete;
+
+  ~Duration();
+
+  /// @brief Returns the duration as microseconds.
+  int64_t microseconds() const { return mgp::duration_get_microseconds(ptr_); }
+
+  bool operator==(const Duration &other) const;
+  Duration operator+(const Duration &other) const;
+  Duration operator-(const Duration &other) const;
+  Duration operator-() const;
+
+  bool operator<(const Duration &other) const;
+
+ private:
+  mgp_duration *ptr_;
+};
+/* #endregion */
+
+/* #endregion */
+
+/* #region Value */
+enum class Type : uint8_t {
+  Null,
+  Bool,
+  Int,
+  Double,
+  String,
+  List,
+  Map,
+  Node,
+  Relationship,
+  Path,
+  Date,
+  LocalTime,
+  LocalDateTime,
+  Duration,
+  UNDEF,  // for indicating an undefined value
```

Comment:
`UNDEF` is only used in `Parameter` objects under `list_item_type`, where it means that the parameter is not a list and thus list item type is left out.
Anyway, I’ll try to avoid this and make a `ListParameter` struct that inherits from `Parameter`, eliminating the need for one type of objects to handle both lists and non-lists.

Reply:
Removed